sklearn/decomposition/nmf.py

```diff
@@ -313,229 +846,108 @@ class ProjectedGradientNMF(BaseEstimator, TransformerMixin):
-           [ 0.38526873,  0.38228063]])
-    >>> model.reconstruction_err_ #doctest: +ELLIPSIS
-    0.00746...
-    >>> model = ProjectedGradientNMF(n_components=2,
-    ...              sparseness='components', init='random', random_state=0)
+    >>> from sklearn.decomposition import NMF
+    >>> model = NMF(n_components=2, init='random', random_state=0)
     >>> model.fit(X) #doctest: +ELLIPSIS +NORMALIZE_WHITESPACE
-    ProjectedGradientNMF(beta=1, eta=0.1, init='random', max_iter=200,
-                n_components=2, nls_max_iter=2000, random_state=0,
-                sparseness='components', tol=0.0001)
+    NMF(alpha=0.0, beta=1, eta=0.1, init='random', l1_ratio=0.0, max_iter=200,
+      n_components=2, nls_max_iter=2000, random_state=0, shuffle=False,
+      solver='coordinate', sparseness=None, tol=0.0001, verbose=0)
+
     >>> model.components_
-    array([[ 1.67481991,  0.29614922],
-           [ 0.        ,  0.4681982 ]])
+    array([[ 2.09783018,  0.30560234],
+           [ 2.13443044,  2.13171694]])
     >>> model.reconstruction_err_ #doctest: +ELLIPSIS
-    0.513...
+    0.00115993...
 
     References
     ----------
-    This implements
-
-    C.-J. Lin. Projected gradient methods
-    for non-negative matrix factorization. Neural
-    Computation, 19(2007), 2756-2779.
+    C.-J. Lin. Projected gradient methods for non-negative matrix
+    factorization. Neural Computation, 19(2007), 2756-2779.
     http://www.csie.ntu.edu.tw/~cjlin/nmf/
-
-    P. Hoyer. Non-negative Matrix Factorization with
-    Sparseness Constraints. Journal of Machine Learning
-    Research 2004.
-
-    NNDSVD is introduced in
-
-    C. Boutsidis, E. Gallopoulos: SVD based
-    initialization: A head start for nonnegative
-    matrix factorization - Pattern Recognition, 2008
-    http://tinyurl.com/nndsvd
     """
 
-    def __init__(self, n_components=None, init=None, sparseness=None, beta=1,
-                 eta=0.1, tol=1e-4, max_iter=200, nls_max_iter=2000,
-                 random_state=None):
+    def __init__(self, n_components=None, solver='coordinate', init=None,
+                 tol=1e-4, max_iter=200, random_state=None,
+                 alpha=0., l1_ratio=0., verbose=0, shuffle=False,
+                 nls_max_iter=2000, sparseness=None, beta=1, eta=0.1):
         self.n_components = n_components
+        self.solver = solver
         self.init = init
         self.tol = tol
-        if sparseness not in (None, 'data', 'components'):
-            raise ValueError(
-                'Invalid sparseness parameter: got %r instead of one of %r' %
-                (sparseness, (None, 'data', 'components')))
+        self.max_iter = max_iter
+        self.random_state = random_state
+        self.alpha = alpha
+        self.l1_ratio = l1_ratio
+        self.verbose = verbose
+        self.shuffle = shuffle
+
+        if sparseness is not None:
+            warnings.warn("Sparseness enforcement in NMF is only available for"
```

Comment:
Rather than "sparseness enforcement" this should say "Controlling regularization through the `sparseness`, `beta` and `eta` arguments"

Enforcement was the wrong word in this context to begin with, my bad.